monitoring/nethealth.go

```diff
@@ -331,12 +296,14 @@ func nethealthFailureProbe(name, peer string, packetLoss float64) *pb.Probe {
 
 // fetchNethealthMetrics collects the network metrics from the nethealth pod
 // specified by addr. Returns the resp as an array of bytes.
-func fetchNethealthMetrics(ctx context.Context, addr string) ([]byte, error) {
-	client, err := roundtrip.NewClient(addr, "")
-	if err != nil {
-		return nil, trace.Wrap(err, "failed to connect to nethealth service at %s.", addr)
+func (c *nethealthChecker) fetchNethealthMetrics(ctx context.Context) (res []byte, err error) {
+	client := http.Client{
+		Transport: &http.Transport{
+			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+				return net.Dial("unix", nethealth.DefaultNethealthSocket)
```

Comment:
instead of using the package global, I'd have nethealthChecker has the socket path as configuration and set it to `DefaultNethealthSocket` if unspecified.